# origin: Debian, #159772
# cset: n/a
# inclusion: not submitted
# description: fix umsdos_solve_hlink oops in dir.c
# revision date: 2004-09-03

diff -urN kernel-source-2.4.26/fs/umsdos/dir.c kernel-source-2.4.26-1/fs/umsdos/dir.c
--- kernel-source-2.4.26/fs/umsdos/dir.c	2003-06-14 00:51:37.000000000 +1000
+++ kernel-source-2.4.26-1/fs/umsdos/dir.c	2003-06-01 20:43:54.000000000 +1000
@@ -688,9 +688,14 @@
 	struct dentry *dentry_dst;
 	char *path, *pt;
 	int len;
-	struct address_space *mapping = hlink->d_inode->i_mapping;
+	struct address_space *mapping;
 	struct page *page;
 
+	dentry_dst = ERR_PTR(-ENOENT);
+	if (!hlink->d_inode)
+		goto out;
+
+	mapping = hlink->d_inode->i_mapping;
 	page=read_cache_page(mapping,0,(filler_t *)mapping->a_ops->readpage,NULL);
 	dentry_dst=(struct dentry *)page;
 	if (IS_ERR(page))
